### PR TITLE
Added a simpler API for timers.

### PR DESCRIPTION
**Pull request by TurkeyMan**

### CODE REVIEW

File: src/urt/time.d [hunk 1]
```diff
@@ -154,6 +154,53 @@ struct Duration
         => cast(double)this;
 }
 
+struct Timer(Duration timeout = Duration.zero)
+{
+nothrow @nogc:
+    MonoTime startTime;
```
Comment (megaeels) [inline src/urt/time.d:160]:
```suggestion
    MonoTime m_startTime;
```

File: src/urt/time.d [hunk 2]
```diff
@@ -154,6 +154,53 @@ struct Duration
...
+    }
+
+    void reset()
+    {
+        startTime = getTime();
+    }
+    void reset(MonoTime now)
+    {
+        startTime = now;
+    }
```
Comment (megaeels) [inline src/urt/time.d:181]:
```suggestion
    void reset(MonoTime now = getTime())
    {
        startTime = now;
    }
```

File: src/urt/time.d [hunk 3]
```diff
@@ -154,6 +154,53 @@ struct Duration
...
+
+    bool expired() const
+        => getTime() - startTime >= m_timeout;
+    bool expired(MonoTime now) const
+        => now - startTime >= m_timeout;
```
Comment (megaeels) [inline src/urt/time.d:186]:
```suggestion
    bool expired(MonoTime now = getTime()) const
        => now - startTime >= m_timeout;
```

File: src/urt/time.d [hunk 4]
```diff
@@ -154,6 +154,53 @@ struct Duration
...
+
+    Duration elapsed() const
+        => getTime() - startTime;
+    Duration elapsed(MonoTime now) const
+        => now - startTime;
```
Comment (megaeels) [inline src/urt/time.d:191]:
```suggestion
    Duration elapsed(MonoTime now = getTime()) const
        => now - startTime;
```

File: src/urt/time.d [hunk 5]
```diff
@@ -154,6 +154,53 @@ struct Duration
...
+
+    Duration remaining() const
+        => m_timeout - (getTime() - startTime);
+    Duration remaining(MonoTime now) const
+        => m_timeout - (now - startTime);
```
Comment (megaeels) [inline src/urt/time.d:196]:
```suggestion
    Duration remaining(MonoTime now = getTime()) const
        => m_timeout - (now - startTime);
```

File: src/urt/time.d [hunk 6]
```diff
@@ -154,6 +154,53 @@ struct Duration
...
+
+    Duration expiredDuration() const
+        => -remaining();
+    Duration expiredDuration(MonoTime now) const
+        => -remaining(now);
```
Comment (megaeels) [inline src/urt/time.d:201]:
```suggestion
    Duration expiredDuration(MonoTime now = getTime()) const
        => -remaining(now);
```

File: src/urt/time.d (hunk 2 above)
Comment (TurkeyMan) [reply src/urt/time.d:181]: So obvious! >_<

File: src/urt/time.d (hunk 1 above)
Comment (TurkeyMan) [reply src/urt/time.d:160]: Yeah... I've been pondering this for a while.

**Pull request merged**